enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,649 @@
+applicable rh-api (for MS).
+
+3. The customer then updates the install-config to turn on the custom-dns
+feature. They also provide the LB Names and types for the pre-created API and
+API-Int LBs. The Installer would use these LB Names to configure them.
+
+4. The Installer skips configuring the cloud DNS solution when this feature is
+enabled via install-config. Since API and API-Int resolution are pre-requisties
+for successful cluster installation, the Installer performs additional
+validations to make sure they are configured correctly. The installer provides
+a warning but does not stop installation when API or API-Int resolution fails.
+
+5. The Installer creates the [DNS CR](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_dns.crd.yaml) without providing values for privateZone
+and publicZone. This indicates to the Ingress Operator to skip configuring the
+cloud DNS with an entry for *.apps.
+
+6. The Installer also sets a field in the Infrastructure CR indicating that
+the custom DNS feature is enabled. This is a way to embed the knowledge of
+this configuration in a running cluster so that this information is available
+during an upgrade too.
+
+7. The Ingress Operator creates and manages the LB for *.apps by starting a k8s
+Service of type LoadBalancer (existing functionality). Updating this Service to
+accept already created LBs is an upstream change that we are not pursuing at
+this time.
+
+8. When the LoadBalancer Service successfully creates the Ingress LB in the
+underlying cloud, it updates the `Hostname` (in AWS) or `IP` (Azure and GCP)
+fields within the [LoadBalancerIngress struct](https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4554). These fields represent the hostname
+or the IP address of the Ingress LB.
+
+9. OpenShift needs to provide an in-cluster DNS solution for *.apps resolution
+because several operators depend on this service being available during cluster
+installation. To achieve this, MCO will start a static CoreDNS pod to provide
+DNS resolution for *.apps only.
+
+10. Configuration for the CoreDNS pod is provided by its CoreFile that contains
+the IP address for the Ingress LB. This IP address is obtained by resolving the
+`Hostname` field within the LB service status.
+
+11. runtimecfg monitors the LB service status for changes to the LB IP address
+(by resolving the `Hostname`). If there is a change, it will update the
+Corefile with the new IP address of the Ingress LB.
+
+12. Once the cluster installation is complete, the customer can add a DNS entry
+in their custom DNS solution for *.apps.
+
+13. During cluster deletion, OpenShift is responsible only for deleting the
+Ingress LB and the customer is responsible for deleting the LBs for API and
+API-Int. The customer would also be responsible for removing entries from
+their custom DNS.
+
+#### Variation [optional]
```

Comment:
@patrickdillon @zaneb, @Miciah and @bennerv , I have updated this section to include information about `spec.loadBalancerIP` and the annotations that have been added by AWS, Azure and GCP that we could potentially use towards the design goal of pre-configuring the DNS entry for *.apps before cluster installation. 

Could you please take a look and provide comments?